### PR TITLE
Add ranger (Issue #438)

### PR DESCRIPTION
**Pull request by mnwright**
Add support for ranger: Random forests for classification, regression and survival.

### CODE REVIEW

**Comment by mllg** [comment]
Code looks solid, only a small note in NEWS is missing.

If there are no objections against the switch to version 4.8 of the gnu compilers (all tests are - somewhat surprisingly - passing), I opt to merge and include in the next release.

File: R/RLearner_classif_ranger.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    properties = c("twoclass", "multiclass", "prob", "numerics", "factors"),
+    name = "Random Forests",
+    short.name = "ranger",
+    note = ""
+  )
+}
+
+#' @export
+trainLearner.classif.ranger <- function(.learner, .task, .subset, .weights, ...) {
+  args = list(...)
```
Comment (mllg) [inline R/RLearner_classif_ranger.R]: Unnecessary.

File: R/RLearner_regr_ranger.R [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    properties = c("numerics", "factors"),
+    name = "Random Forests",
+    short.name = "ranger",
+    note = ""
+  )
+}
+
+#' @export
+trainLearner.regr.ranger = function(.learner, .task, .subset, .weights, ...) {
+  args = list(...)
```
Comment (mllg) [inline R/RLearner_regr_ranger.R]: Unnecessary.

File: R/RLearner_surv_ranger.R [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    properties = c("numerics", "factors", "rcens", "prob"),
+    name = "Random Forests",
+    short.name = "ranger",
+    note = ""
+  )
+}
+
+#' @export
+trainLearner.surv.ranger = function(.learner, .task, .subset, .weights, ...) {
+  args = list(...)
```
Comment (mllg) [inline R/RLearner_surv_ranger.R]: Unnecessary.

**Comment by jakob-r** [comment]
If I am not mistaken `importance` also should have the `tunable = FALSE` flag, or am I wrong?

**Comment by berndbischl** [comment]
importance: if this just a flag to do the calculation and has no effect on model quality it should be tunable = F.

@all:
We dont merge these thing from now on, before they are squashed in the PR. So please do this.

**Comment by jakob-r** [comment]
@berndbischl Here I would be very carefull with that as there has been a "merge branch master into add_ranger".

**Comment by berndbischl** [comment]
This cannot be so hard. We add basically 6 new files, 3 for learners and 3 for tests. 
And some lines in the travis.yml and news. I dont want 10 commits in the master history for this. 
Please do this.

**Comment by berndbischl** [comment]
Thx for squashing. But should we set importance and the scale.imporatance param to tunable = FALSE?

**Comment by jakob-r** [comment]
I think the squashing didn't give any clearity to this pull request (see Files Changed etc) :unamused:

**Comment by berndbischl** [comment]
I would guess that @mllg merged on the console. The master commit history looks good. But it would be nice if we would write here what happened.

**Comment by mllg** [comment]
I simply merged the conflicts in `.travis.yml`. `tunable = FALSE` is now also set for the importance stuff.

**Pull request merged**